### PR TITLE
[Feature][Transform] Add embedding transform

### LINKED ISSUE #7526: [Feature][Transform] Add embedding transform
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Add embedding transform to fulfill the requirement of converting data to vectors

Reference config：
```
transform {
  Embedding {
    source_table_name = "fake"
    embedding_model_provider = QIANFAN
    model = bge_large_en
    api_key = xxxxxxx
    secret_key = xxxxxxx
    api_path = "https://aip.baidubce.com/rpc/2.0/ai_custom/v1/wenxinworkshop/embeddings"
    vectorization_fields {
        book_intro_vector = book_intro  # source field = taget field  , refer to copy transform
    }
    result_table_name = "embedding_output"
  }
}
```

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by corgy-w**
### Purpose of this pull request
issue : #7526 

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: docs/en/transform-v2/embedding.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
+# Embedding
```
Comment (Hisoka-X) [inline docs/en/transform-v2/embedding.md:1]: How about renamed to `EmbeddingRemote`? So that we can support `EmbeddingLocal` in the future.

Comment (corgy-w) [reply docs/en/transform-v2/embedding.md:1]: get

File: docs/en/transform-v2/embedding.md [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+| oauth_path                     | string | no       | -             | The API endpoint for the oauth service.                                                                     |
+| custom_config                  | map    | no       |               | Custom configurations for the model.                                                                        |
+| custom_response_parse          | string | no       |               | Specifies how to parse the response from the model using JsonPath. Example: `$.choices[*].message.content`. |
+| custom_request_headers         | map    | no       |               | Custom headers for the request to the model.                                                                |
+| custom_request_body            | map    | no       |               | Custom body for the request. Supports placeholders like `${model}`, `${input}`, `${prompt}`.                |
+
+### model_provider
+
+The model provider to use for generating embeddings. Common options might include `QIANFAN`, `OPENAI`, etc. Depending on
```
Comment (Hisoka-X) [inline docs/en/transform-v2/embedding.md]: Why not list all supported model?

File: docs/zh/transform-v2/embedding.md [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+
+用于向oauth服务发送请求的API,获取对应的认证信息。根据提供商和所用模型的不同可能有所变化。通常由模型提供商提供。
+
+### custom_config
+
+`custom_config` 选项允许您为模型提供额外的自定义配置。这是一个映射，您可以在其中定义特定模型可能需要的各种设置。
+
+### custom_response_parse
+
+`custom_response_parse` 选项允许您指定如何解析模型的响应。您可以使用 JsonPath
```
Comment (Hisoka-X) [inline docs/zh/transform-v2/embedding.md:72]: How about link to jsonpath? So user can know how to configure it.

File: docs/en/transform-v2/embedding.md (hunk 2 above)
Comment (corgy-w) [reply docs/en/transform-v2/embedding.md]: get

**Review by Hisoka-X — DISMISSED** [review]
LGTM. Thanks @corgy-w !

**Review by Hisoka-X — APPROVED** [review]

**Review by hailin0 — APPROVED** [review]

**Pull request merged**